src/main/java/com/monetate/koupler/format/SplitFormat.java

```diff
@@ -0,0 +1,34 @@
+package com.monetate.koupler.format;
+
+import org.apache.commons.cli.CommandLine;
+
+/**
+ * Created by brianoneill on 4/29/16.
+ */
+public class SplitFormat implements Format {
+    private int partitionKeyField = 0;
+    private String delimiter = ",";
+
+    /**
+     * Constructor for Split format
+     */
+    public SplitFormat(CommandLine cmd){
+        if (cmd.hasOption("delimiter")) {
+            delimiter = cmd.getOptionValue("delimiter");
+        }
+
+        if (cmd.hasOption("partitionKeyField")) {
+            partitionKeyField = Integer.parseInt(cmd.getOptionValue("partitionKeyField"));
+        }
+    }
+
+    @Override
+    public String getPartitionKey(String event) {
+        return event.split(this.delimiter)[partitionKeyField];
+    }
+
+    @Override
+    public String getData(String event) {
+        return event;
```

Comment:
yeah -- the split format is a case where you are processing lines of a log file (for example)    syslog --> koupler (via UDP) for instance.

You want to send the log line in its entirety as the payload, and use a certain field out of it as the partition key.  (maybe app name... so all logs for a specific app go to the same shard / consumer)